subprojects/test-runner/robolectric-inhouse-runner/src/main/kotlin/com/avito/robolectric/runner/description/RobolectricDescriptionMetaDataParser.kt

```diff
@@ -0,0 +1,23 @@
+package com.avito.robolectric.runner.description
+
+import com.avito.test.report.listener.description.DescriptionMetaData
+import com.avito.test.report.listener.description.DescriptionMetadataParser
+import org.junit.runner.Description
+
+internal class RobolectricDescriptionMetaDataParser : DescriptionMetadataParser {
+
+    override fun parse(desc: Description): DescriptionMetaData {
+        val result = REGEX.find(desc.displayName)
+            ?: throw IllegalArgumentException("Cannot parse description: ${desc.displayName}")
+
+        return DescriptionMetaData(
+            environment = "Robolectric-${result.groupValues[2]}",
+            testName = result.groupValues[1],
+            className = result.groupValues[3]
+        )
+    }
+
+    private companion object {
+        private val REGEX = Regex("([\\s\\S]*)\\[(.*)]\\((.*)\\)")
```

Comment:
Using `.*` between two delimiters is greedy and will capture text between multiple pairs.
E.g. regex `\[.+\]` in `[a] text between [b] text between [c]` string will match the entire test as only one match.

If you need to capture test only between delimiters, you should exclude delimiter by `[^delimiter]` instead of using any symbol - `.`.
For the previous example, `\[[^\]]+\]` will give three matches - `[a]`, `[b]`, `[c]`.

If we don't control the count of pairs ("[22]" as I see), it might be safer to find them all and filter unneeded.